internal/cache/lru/lru.go

```diff
@@ -151,7 +151,8 @@ func (c *Cache) Insert(
 		return nil, errors.New(InvalidEntryErrorMsg)
 	}
 
-	if value.Size() > c.maxSize {
+	valueSize := value.Size()
```

Comment:
This code change is remove the unnecessary duplicate Size() calls on the new entry being inserted.